### PR TITLE
Fix wireguard test for sle

### PR DESCRIPTION
**Pull request by asdil12**
Ticket: https://jira.suse.com/browse/SLE-14409
Verification: http://kazhua.qa.suse.de/tests/208 (unrelated fail due to only one node being executed)

### CODE REVIEW

**Review by pdostal — APPROVED** [review]

**Comment by pdostal** [comment]
The code looks good, but please restart the VR ;)

**Comment by DimStar77** [comment]
Just curious: will this help with stability of the test on TW?

https://openqa.opensuse.org/tests/1624459#next_previous - This test keeps on requiring multiple runs to pass

**Review by grisu48 — APPROVED** [review]
Cool! Thanks for working on this. Looking forward to the VR passing

**Comment by grisu48** [comment]
> Just curious: will this help with stability of the test on TW?
> 
> https://openqa.opensuse.org/tests/1624459#next_previous - This test keeps on requiring multiple runs to pass

I created https://progress.opensuse.org/issues/88511 for it. @pdostal  and me can have a look in the next weeks when we have some time.

**Comment by DimStar77** [comment]
> I created https://progress.opensuse.org/issues/88511 for the instability issue. @pdostal and me can have a look in the next weeks when we have some time.

Thanks; much appreciated!

**Comment by asdil12** [comment]
> Cool! Thanks for working on this. Looking forward to the VR passing

I can't make it pass as my setup doesn't support multi machine.
That's why this test runs in the devel jobgroup to see if it passes there.

File: tests/network/wireguard.pm [hunk 1]
```diff
@@ -58,6 +60,21 @@ sub run {
...
+        add_suseconnect_product('sle-module-desktop-applications');
+        script_run('source /etc/os-release');
+        # Workaround https://bugzilla.suse.com/show_bug.cgi?id=1181941
+        my $scn = script_run('SUSEConnect -p sle-we/${VERSION_ID}/${CPU} -r ' . get_required_var('SCC_REGCODE_WE'));
+        die if ($scn ne 0 && $scn ne 106);
+        zypper_call '--gpg-auto-import-keys ref';
+        # Workaround https://bugzilla.suse.com/show_bug.cgi?id=1182004
+        $scn = script_run('SUSEConnect -p PackageHub/${VERSION_ID}/${CPU}');
+        die if ($scn ne 0 && $scn ne 106);
+        zypper_call '--gpg-auto-import-keys ref';
```
Comment (foursixnine) [inline tests/network/wireguard.pm:73]: shouldn't add_suseconnect_product take care of both cases?

Comment (asdil12) [reply tests/network/wireguard.pm:73]: It doesn't. That's what I needed to find out today step by step. At least if there is no undocumented option that allows this.

Comment (pdostal) [reply tests/network/wireguard.pm:73]:
```perl
add_suseconnect_product('PackageHub', undef, undef, undef, 300, 1)
```

Comment (asdil12) [reply tests/network/wireguard.pm:73]: According to the code this shouldn't help:

```perl
    my $result = script_run("SUSEConnect -p $name/$version/$arch $params", $timeout);
    if ($result != 0 && $retry) {
```

This retry option would help to workaround https://bugzilla.suse.com/show_bug.cgi?id=1124318 (which got resolved in the meantime) but the problem now is that SUSEConnect doesn't run into an error but also doesn't import the repo GPG key: https://bugzilla.suse.com/show_bug.cgi?id=1182004

I think I can replace the PackageHub `script_run` call with `add_suseconnect_product` but I will need to keep the `zypper_ref` call to import the gpg key of the PackageHub repo.

Comment (asdil12) [reply tests/network/wireguard.pm:73]: The WE module actually seems to still need the retry option - and also the `zypper --gpg-auto-import-keys ref` as well.

Comment (asdil12) [reply tests/network/wireguard.pm]: Updated PR and verification run.

**Comment by pdostal** [comment]
@asdil12 did you do at least one successful verification run?

**Comment by asdil12** [comment]
http://kazhua.qa.suse.de/tests/208 ran until the it tried to interact with the other multimachine node - which - as I stated in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11928#issuecomment-776024859 - my setup doesn't support.
Therefore this job is only scheduled in the sle devel jobgroup atm. As soon as it works, I will move it to the main jobgroup.

**Comment by pdostal** [comment]
> http://kazhua.qa.suse.de/tests/208 ran fine until the it tried to interact with the other multimachine node - which - as I stated in [#11928 (comment)](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11928#issuecomment-776024859) - my setup doesn't support.
> Therefore this job is only scheduled in the sle devel jobgroup atm. As soon as it works, I will move it to the main jobgroup.

Can you put here the link to the SLE jobgroup?

**Comment by asdil12** [comment]
> Can you put here the link to the SLE jobgroup?

Currently it runs in the test development jobgroup: https://openqa.suse.de/group_overview/96
https://openqa.suse.de/tests/5451356

As it failed sporadically, I pinned the wireguard tests to a specific worker (maybe the tap network doesn't work across workers here?) and will observe if the stability increases.

**Comment by pdostal** [comment]
Thank you @asdil12, the last run is green! We will see! :)

**Pull request merged**